programs/clearing_house/src/math/margin.rs

```diff
@@ -283,6 +284,8 @@ pub fn calculate_margin_requirement_and_total_collateral_and_liability_info(
     };
 
     for spot_position in user.spot_positions.iter() {
+        validate_spot_position(spot_position)?;
```

Comment:
might be nicer to add these in place/fill/trigger/cancel instead?

Reply:
feel like theres validates to try to prevent (like the list above) and validates to try and detect (in this bottlenecked area)